sentry-types/src/protocol/envelope.rs

```diff
@@ -112,12 +116,63 @@ pub enum EnvelopeItem {
     Attachment(Attachment),
     /// A MonitorCheckIn item.
     MonitorCheckIn(MonitorCheckIn),
+    /// A container for a list of multiple items.
+    ItemContainer(ItemContainer),
     /// This is a sentinel item used to `filter` raw envelopes.
     Raw,
     // TODO:
     // etc…
 }
 
+/// A container for a list of multiple items.
+/// It's considered a single envelope item, with its `type` corresponding to the contained items'
+/// `type`.
+#[derive(Clone, Debug, PartialEq)]
+pub enum ItemContainer {
```

Comment:
marking this as `#[non_exhaustive]` means we can add more variants in the future without breaking semver